app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -225,21 +225,25 @@ class GeneralApplicationMongoRepository(
         val applicationStatus = document.getAs[ApplicationStatus]("applicationStatus").get
         val applicationRoute = document.getAs[ApplicationRoute]("applicationRoute").getOrElse(ApplicationRoute.Faststream)
         val progressStatusTimeStampDoc = document.getAs[BSONDocument]("progress-status-timestamp")
-        val progressStatusTimeStamp = progressStatusTimeStampDoc.flatMap { timestamps =>
+        val latestProgressStatus = progressStatusTimeStampDoc.flatMap { timestamps =>
           val relevantProgressStatuses = timestamps.elements.filter(_._1.startsWith(applicationStatus))
+          val latestRelevantProgressStatus = relevantProgressStatuses.maxBy(element => timestamps.getAs[DateTime](element._1).get)
+          Try(ProgressStatuses.nameToProgressStatus(latestRelevantProgressStatus._1)) match {
+            case Success(progressStatus) => Some(progressStatus)
```

Comment:
You can use `.toOption` function on `Try()`

Reply:
:+1: